lib/reply.js

```diff
@@ -666,23 +669,58 @@ function sendStream (payload, res, reply) {
   } else {
     reply.log.warn('response will send, but you shouldn\'t use res.writeHead in stream mode')
   }
-  payload.pipe(res)
+  if (reply[kReplyTrailers] === null) {
+    stream.pipe(res)
+  } else {
+    // we should not pipe to res when using trailer
+    // it will immediately close the response after
+    // the payload is drain
+    stream.on('data', function (chunk, encoding) {
+      console.log(chunk, encoding)
+      res.write(chunk, encoding)
+    })
```

Comment:
This will not respect backpressure, we should use `pipe`.

Reply:
This is not possible to use `pipe`. Using `pipe` will immediately end the response after the `payload` is drain.
I cannot found a proper option for it.

Reply:
Strange, after some refactoring. It does work with `pipe`.

Reply:
You can use `stream.pipe(res, { end: false })`.